src/core/server/plugins/plugins_system.ts

```diff
@@ -57,29 +57,31 @@ export class PluginsSystem {
 
       this.log.debug(`Setting up plugin "${pluginName}"...`);
 
-      const exposedDependencyValues = [
-        ...plugin.requiredDependencies,
-        ...plugin.optionalDependencies,
-      ].reduce(
-        (dependencies, dependencyName) => {
-          dependencies[dependencyName] = exposedValues.get(dependencyName);
-          return dependencies;
+      const pluginDepContracts = [...plugin.requiredPlugins, ...plugin.optionalPlugins].reduce(
+        (depContracts, dependencyName) => {
+          // Only set if present. Could be absent if plugin does not have server-side code or is a
+          // missing optional dependency.
+          if (contracts.get(dependencyName) !== undefined) {
```

Comment:
Instead of having duplicate `get` calls, can we do a `has` check instead, i.e., will that still catch this `undefined` case?

```js
if (contracts.has(dependencyName)) {
  depContracts[dependencyName] = contracts.get(dependencyName);
}
```